### PR TITLE
Address more feedback on USM PR

### PR DESCRIPTION
**Pull request by muhammad-tanvir-1211**
This patch includes the following:

* Library compilation fix when using `index_t = long`
* Replace `is_dpcpp` with `SB_ENABLE_USM` in `CMakeFunctionHelper.cmake` file
* Update README to add USM specific information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,14 +175,18 @@ of multiple BLAS operations.
 
 ## API description
 
-This section references all the supported operations and their interface.
+This section references all the supported operations and their interface. The library follows the [oneAPI MKL BLAS specification](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/blas/blas.html) as reference for the api. We have support for both USM and Buffer api, however the group apis for USM are not supported. We don't support mixing USM and Buffer arguments together to compile the library, and instead stick to the aformentioned reference specification.
```
Comment (pgorlani) [inline README.md:178]: It would be better to add a couple of newlines for padding it with the rest of the text.

**Review by pgorlani — APPROVED** [review]

File: src/interface/blas2_interface.hpp [hunk 2]
```diff
@@ -745,8 +745,8 @@ typename sb_handle_t::event_t _tbsv_impl(
   auto sync_buffer = blas::helper::allocate < is_usm
                          ? blas::helper::AllocType::usm
                          : blas::helper::AllocType::buffer,
-       index_t > (sync_vec.size(), queue);
-  auto copy_sync = blas::helper::copy_to_device<index_t>(
+       int32_t > (sync_vec.size(), queue);
+  auto copy_sync = blas::helper::copy_to_device<int32_t>(
```
Comment (aacostadiaz) [inline src/interface/blas2_interface.hpp:749]: Why is this change needed?

**Review by aacostadiaz — APPROVED** [review]
Looks good! I just left a minor comment.

File: src/interface/blas2_interface.hpp (hunk 2 above)
Comment (aacostadiaz) [reply src/interface/blas2_interface.hpp:749]: Nevermind, you explained that in the PR description

File: README.md (hunk 1 above)
Comment (muhammad-tanvir-1211) [reply README.md]: Addressed this in the latest commit.
